presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -156,6 +160,46 @@ void registerQuery(
         }
     }
 
+    public synchronized DynamicFiltersStats getDynamicFilteringStats(QueryId queryId, ConnectorSession session)
```

Comment:
This synchronized scope may be a little broader than needed. If many queries using dynamic filtering run concurrently, this part to get the dynamic filtering stats can cause the long waiting. 

Looks like we may need to guard only `dynamicFilterSummaries`?

Reply:
agreed, made changes to restrict guard to smaller scope